sklearn/covariance/_empirical_covariance.py

```diff
@@ -84,7 +84,7 @@ def empirical_covariance(X, *, assume_centered=False):
            [0.25, 0.25, 0.25],
            [0.25, 0.25, 0.25]])
     """
-    X = np.asarray(X)
+    X = check_array(X, ensure_2d=False)
```

Comment:
Thinking about it more, there are quite a few places where we call `empirical_covariance`, which means a lot of `with config_context(...)`. Since we did not check for finite-ness before, I am okay with not checking it now:

```suggestion
    X = check_array(X, ensure_2d=False, force_all_finite=False)
```

This way we do not need to add `config_context` everywhere.